src/seedu/addressbook/commands/fees/EditFeesCommand.java

```diff
@@ -33,6 +33,12 @@
      */
     public EditFeesCommand(int index, String fees, String date) throws IllegalValueException {
         setTargetIndex(index, ObjectTargeted.PERSON);
+        if (!isValidDate(date) && !"0".equals(date)) {
+            throw new IllegalValueException(MESSAGE_DATE_CONSTRAINTS);
+        }
+        if ("0.00".equals(fees)) {
+            throw new IllegalValueException(MESSAGE_FEES_VALUE_CONSTRAINTS);
+        }
         this.fees = new Fees(fees, date);
     }
```

Comment:
This looks quite potato. Cause you'll allow things like "Garbage" as your fees.
fees should be a double which you convert from String to double.
Then you can also check if it is converted properly and that it is in the $ format, so not things like $0.3333.
If it isn't formatted properly, throw a error message
But it'll require quite a lot of changes to make it clean though

Reply:
Actually the regex alr checks for correct values as Strings, if u look under the Fees class. It's just that idk how to make 1 exception for the 0 value, that's why this is here.

So things like "Garbage" or "0.33333333" won't go through.....

Reply:
It still weird that your fees value is stored as a string instead of a number data type

Reply:
Had this conversation w Mick early on during the Sem and I forgot what we talked about but we agreed on using Strings

Reply:
Fine, I yield